src/Features/Core/Portable/PullMemberUp/AbstractPullMemberUpRefactoringProvider.cs

```diff
@@ -39,50 +40,66 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return;
             }
 
-            var selectedMemberNode = await GetSelectedNodeAsync(context).ConfigureAwait(false);
-            if (selectedMemberNode == null)
+            var selectedMemberNodes = await GetSelectedNodesAsync(context).ConfigureAwait(false);
+            if (selectedMemberNodes.IsEmpty)
             {
                 return;
             }
 
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var selectedMember = semanticModel.GetDeclaredSymbol(selectedMemberNode);
-            if (selectedMember == null || selectedMember.ContainingType == null)
+            var memberNodeSymbolPairs = selectedMemberNodes
+                .Select(m => (node: m, symbol: semanticModel.GetDeclaredSymbol(m, cancellationToken)))
+                .Where(pair => pair.symbol != null && MemberAndDestinationValidator.IsMemberValid(pair.symbol))
+                .AsImmutable();
+
+            var selectedMembers = memberNodeSymbolPairs.SelectAsArray(pair => pair.symbol!);
+
+            if (selectedMembers.IsEmpty)
             {
                 return;
             }
 
-            if (!MemberAndDestinationValidator.IsMemberValid(selectedMember))
+            var containingType = selectedMembers.First().ContainingType;
+            if (containingType == null || selectedMembers.Any(static (m, containingType) => m.ContainingType != containingType, containingType))
             {
                 return;
             }
 
+            // we want to use a span which covers all the selected viable member nodes, so that more specific nodes have priority
+            var memberSpan = new SyntaxList<SyntaxNode>(memberNodeSymbolPairs.Select(pair => pair.node)).FullSpan;
+
             var allDestinations = FindAllValidDestinations(
-                selectedMember,
+                selectedMembers,
+                containingType,
                 document.Project.Solution,
                 cancellationToken);
             if (allDestinations.Length == 0)
             {
                 return;
             }
 
-            var allActions = allDestinations.Select(destination => MembersPuller.TryComputeCodeAction(document, selectedMember, destination, context.Options))
-                .WhereNotNull().Concat(new PullMemberUpWithDialogCodeAction(document, selectedMember, _service, context.Options))
+            var allActions = allDestinations.Select(destination => MembersPuller.TryComputeCodeAction(document, selectedMembers, destination, context.Options))
+                .WhereNotNull().Concat(new PullMemberUpWithDialogCodeAction(document, selectedMembers, _service, context.Options))
                 .ToImmutableArray();
 
+            var title = selectedMembers.IsSingle() ?
+                string.Format(FeaturesResources.Pull_0_up, selectedMembers.Single().ToNameDisplayString()) :
+                FeaturesResources.Pull_selected_members_up;
```

Comment:
```suggestion
            var title = selectedMembers.IsSingle()
                ? string.Format(FeaturesResources.Pull_0_up, selectedMembers.Single().ToNameDisplayString())
                : FeaturesResources.Pull_selected_members_up;
```

Allows the `?:` to align, and makes it very easy to immediately see that this is a conditional expr and what the cases will be.